### PR TITLE
Master merged cob command

### LINKED ISSUE #62: Unify msh_upstart and upstart (cob_command)
At the moment each upstart configuration has a different setup.

### LINKED ISSUE #67: Develop a new upstart software
The idea is that upstart will be started  based on a yaml file like:

```
robot: cob4-7
user: msh


pcs:"
$ROBOT-b1
$ROBOT-t1
$ROBOT-t2
$ROBOT-t3
$ROBOT-s1
$ROBOT-h1"

check_pcs:"
$ROBOT-t1
$ROBOT-t3
$ROBOT-s1"

sessions:
  ros:
    roscore:
      command: roscore
      pre_condition:
    bringup: 
      command: roslaunch cob_bringup robot.launch
      pre_condition:  rostopic list
    sm:
      command: roslaunch msh_statemachine machine.launch
      pre_condition:  rosparam get /robot_description
  extra:
    start-vm:
      command: su mimic -c "ssh $ROBOT-h1 'export DISPLAY=:0 && virtualbox --startvm $ROBOT-win --fullscreen'"&
      pre_condition: su mimic -c "ssh $ROBOT-h1 'vboxmanage list runningvms'" | grep $ROBOT-win
```

### PR DESCRIPTION
**Pull request by ipa-fxm**
fix #62 
fix #67
supersedes #76 

This branch consists of latest ipa320/master and then merged with #76 
Merge conflicts have been resolved together with @ipa-nhg in an additional commit `upstart` and `upstart_msh` have been synced and cleaned up, i.e. move common scripts into `scripts` folder...

### CODE REVIEW

**Comment by ipa-fxm** [comment]
I was able to execute `upstart_msh/./upstart_install.sh`...afterwards `cob-command` was available...

Unfortunately, I am not too familiar with its usage:
 - `cob-command -h` shows the help
 - `cob-command ls` starts with a "Listing the sessions" banner...then asks for a password...robot pw does not work
 - `cob-command start` also asks for password and seams to get stuck...

@ipa-nhg let's test the usage together...whether the `cob-command` behavior was not broken by the merge and whether still everything works as expected...

**Comment by ipa-fxm** [comment]
the last three commits have been done offline without testing yet...
I was just looking through the files and try to harmonize them

@ipa-nhg 
Do you think it is possible to replace some `sed myrobot` commands by directly using the `ROBOT` environment variable? With this we could further reduce commands as started in the latest commit...

All of the last three commits need review and testing on the robot...

File: test_scripts/start_test_cam3dASUS.sh [hunk 1]
```diff
@@ -1,24 +0,0 @@
-#!/bin/bash
```
Comment (ipa-nhg) [inline test_scripts/start_test_cam3dASUS.sh]: Please don't remove this file

File: test_scripts/check_cam3dASUS.sh [hunk 2]
```diff
@@ -1,57 +0,0 @@
-#!/bin/bash
-#copy this file to /etc/init.d folder
-#and call "sudo update-rc.d check_cam3dASUS.sh defaults"
-#
-sleep 30
```
Comment (ipa-nhg) [inline test_scripts/check_cam3dASUS.sh]: Please don't remove these files

**Comment by ipa-nhg** [comment]
Please revert https://github.com/ipa320/setup_cob4/pull/92/commits/664ca49df47dc12d60676c777e119bc64dfa7c88 and https://github.com/ipa320/setup_cob4/pull/92/commits/53f99f49ce3dfa612b59a636c1cb64c39f808277

**Comment by ipa-nhg** [comment]
Close this PR and open a new clean one https://github.com/ipa320/setup_cob4/pull/96
See https://github.com/ipa320/setup_cob4/issues/89#issuecomment-305542230 , more commits have to be reverted, I don't have any more an overview of the changes...

**Pull request closed without merging**